libethcore/Transaction.cpp

```diff
@@ -111,10 +112,14 @@ Address const& TransactionBase::sender() const
 {
 	if (!m_sender)
 	{
-		auto p = recover(m_vrs, sha3(WithoutSignature));
-		if (!p)
-			BOOST_THROW_EXCEPTION(InvalidSignature());
-		m_sender = right160(dev::sha3(bytesConstRef(p.data(), sizeof(p))));
+		if (hasZeroSignature())
+			m_sender = MaxAddress;
+		else
+		{
+			auto p = recover(m_vrs, sha3(WithoutSignature));
+			if (p)
```

Comment:
Important: When can `recover()` fail?

If the question is answered, you can do this in single line: `m_sender = recover(...)`. In case of failure `m_sender` will be still "null".